SSA/Projects/InstCombine/LLVM/PrettyEDSL.lean

```diff
@@ -0,0 +1,64 @@
+import SSA.Core.MLIRSyntax.GenericParser
+import SSA.Projects.InstCombine.Tactic
+open Lean
+
+declare_syntax_cat InstCombine.un_op_name
+declare_syntax_cat InstCombine.bin_op_name
+
+syntax "llvm.return"  : InstCombine.un_op_name
+syntax "llvm.copy"    : InstCombine.un_op_name
+syntax "llvm.neg"     : InstCombine.un_op_name
+syntax "llvm.not"     : InstCombine.un_op_name
+
+syntax "llvm.add"     : InstCombine.bin_op_name
+syntax "llvm.and"     : InstCombine.bin_op_name
+syntax "llvm.ashr"    : InstCombine.bin_op_name
+syntax "llvm.lshr"    : InstCombine.bin_op_name
+syntax "llvm.mul"     : InstCombine.bin_op_name
+syntax "llvm.or"      : InstCombine.bin_op_name
+syntax "llvm.sdiv"    : InstCombine.bin_op_name
+syntax "llvm.shl"     : InstCombine.bin_op_name
+syntax "llvm.srem"    : InstCombine.bin_op_name
+syntax "llvm.sub"     : InstCombine.bin_op_name
+syntax "llvm.udiv"    : InstCombine.bin_op_name
+syntax "llvm.urem"    : InstCombine.bin_op_name
+syntax "llvm.xor"     : InstCombine.bin_op_name
+
+-- TODO: does `icmp` need its own case?
+-- TODO: does `select` need its own case?
+
+
+syntax (mlir_op_operand " = ")? InstCombine.un_op_name mlir_op_operand " : " mlir_type : mlir_op
+macro_rules
+  | `([mlir_op| $[$resName =]? $name:InstCombine.un_op_name $x : $t ]) => do
+    let name ← match name.raw with
+      | .node _ _ ⟨.atom _ name :: _⟩ => pure name
+      | _ => Macro.throwErrorAt name s!"Expected an atom, found: {name}" --TODO: better error message
+    let opName := Syntax.mkStrLit name
+    let retTy : TSyntaxArray `mlir_type := match resName with
+      | some _ => #[t]
+      | none => #[]
+    `([mlir_op| $[$resName =]? $opName ($x) : ($t) -> ($retTy:mlir_type,*) ])
+
+macro resName:mlir_op_operand " = " name:InstCombine.bin_op_name
+    x:mlir_op_operand ", " y:mlir_op_operand " : " t:mlir_type : mlir_op  => do
+  let name ← match name.raw with
+    | .node _ _ ⟨.atom _ name :: _⟩ => pure name
+    | _ => Macro.throwErrorAt name s!"Expected an atom, found: {name}" --TODO: better error message
+  let opName := Syntax.mkStrLit name
+  `(mlir_op| $resName:mlir_op_operand = $opName ($x, $y) : ($t, $t) -> ($t) )
+
+set_option hygiene false in
+macro res:mlir_op_operand " = " "llvm.mlir.constant" x:num " : " t:mlir_type : mlir_op =>
```

Comment:
To double check, do we still need hygiene false in this case? Or can this be removed after your recent hygiene-related discoveries?

Reply:
Indeed, this option can be removed now that our idents are no longer affected by hygiene